addons/website_sale_referral/static/src/js/referral_widget.js

```diff
@@ -0,0 +1,173 @@
+odoo.define('website_sale_referral.referral_widget', function (require) {
+    "use strict";
+
+var publicWidget = require('web.public.widget');
+var ajax = require('web.ajax');
+var core = require('web.core');
+var QWeb = core.qweb;
+
+publicWidget.registry.ReferralWidget = publicWidget.Widget.extend({
+    xmlDependencies: ['/website_sale_referral/static/src/xml/referral_tracking_sub_template.xml'],
+    selector:'.referral_widget',
+    events: {
+        'click .share_social_network': 'onclick_share_social_network',
+        'click .get_link' : 'onclick_get_link',
+    },
+
+    start: function() {
+        this.load_tracking();
+        return this._super.apply(this, arguments);
+    },
+
+    load_tracking: function() {
+        var token = $("input[name='referral_token']").val();
+        var url = token ? '/referral/tracking/'.concat(token) : '/referral/tracking/';
+        var self = this;
+        ajax.jsonRpc(url, 'call', {})
+        .then(function (data) {
+            self.currency_symbol = data.currency_symbol;
+            self.currency_position = data.currency_position;
+            self.reward_value = data.reward_value;
+            var referrals = 'my_referrals' in data ? data.my_referrals : {};
+            self.render_tracking(referrals);
+        });
+    },
+
+    render_tracking: function(data) {
+        var referrals = data;
+        this.is_demo_data = false;
+        if(Object.keys(referrals).length == 0) {
+            referrals = this.get_example_referral_statuses();
+            this.currency_symbol = '$';
+            this.currency_position = 'before';
+            this.reward_value = 200;
+            this.is_demo_data = true;
+        }
+        this.referrals_count = Object.keys(referrals).length;
+        var referrals_won = 0;
+        var r;
+        for(r in referrals) {
+            if(referrals[r].state == 'done') {
+                referrals_won++;
+            }
+        }
+        var context = {
+            'my_referrals': referrals,
+            'reward_value': this.reward_value,
+            'total_reward': this.reward_value_to_text(referrals_won),
+            'potential_reward': this.reward_value_to_text(this.referrals_count)
+        };
+        var rendered_html = QWeb.render('referral_tracking_sub_template', context);
+        if(this.is_demo_data) {
+            rendered_html = "<div class='o_sample_overlay bg-white'/>".concat(rendered_html);
+        }
+        $("div[id='referral_tracking_sub_template']").html(rendered_html);
+    },
+
+    reward_value_to_text: function(quantity) {
+        if(this.currency_position == 'after') {
+            return (quantity * this.reward_value).toString().concat(this.currency_symbol);
+        }
+        else {
+            return this.currency_symbol.concat((quantity * this.reward_value).toString());
+        }
+    },
+
+    get_example_referral_statuses: function() {
```

Comment:
Whut ?

Reply:
If this is what I think, this should be removed ^^

Reply:
This is not test data, nor demo data. When no referral has been created yet we display sample data (see https://drive.google.com/file/d/1wkvfC85q49NRXne9vml63J9f1lg-sFqk/view?usp=sharing). This is that data.

I renamed the function to make it more clear.

I'm not sure a script is the best place to put this but this is the only place this is used, so placing it somewhere else would imply overhead to get the data here.

I'm open to any suggestion of course.